build/dependency

```diff
@@ -25,23 +25,54 @@ Examples:
 EOF
 }
 
+pkg_chdir() {
+    #
+    # Note: we must have three layers of directory because old autotools
+    # misbehave when we are using only two layers and specifically they try
+    # to use measurement-kit's autotools rather than their own.
+    #
+    # See measurement-kit/measurement-kit#1017 build history for more.
+    #
+    cd ./build/third_party/$pkg_name
+}
+
 pkg_autogen() {
     if [ -x ./autogen.sh ]; then
-        echo "- ./autogen.sh"
-        ./autogen.sh
+        echo "- ./autogen.sh $pkg_autogen_sh_flags"
+        ./autogen.sh $pkg_autogen_sh_flags
     elif [ -f ./configure.ac ]; then
         echo "- autoreconf -i"
         autoreconf -i
     fi
 }
 
+pkg_depend() {
+    for dep in `echo $pkg_deps`; do
+        if [ "$pkg_cross" != "" -a "$pkg_arch" != "" ]; then
+            dep="$pkg_cross-$pkg_arch-$dep"
+        fi
+        if [ ! -f $pkg_rootdir/$dep.tgz ]; then
+            baseurl=https://github.com/measurement-kit/dependencies
+            wget -O $pkg_rootdir/$dep.tgz                                      \
+                $baseurl/releases/download/2016-12-19/$dep.tgz
+            wget -O $pkg_rootdir/$dep.tgz.asc                                  \
+                $baseurl/releases/download/2016-12-19/$dep.tgz.asc
+            gpg2 --verify $pkg_rootdir/$dep.tgz.asc
+            install -d $pkg_prefix
+            tar -C $pkg_prefix -xzf $pkg_rootdir/$dep.tgz
+        else
+            echo "It seems $dep has already been built and installed"
+        fi
+    done
+}
+
 pkg_configure() {
     if [ -x ./configure ]; then
         echo "- ./configure --prefix=$pkg_prefix $pkg_configure_flags"
-        ./configure --prefix=$pkg_prefix $pkg_configure_flags
+        ./configure --prefix=/ $pkg_configure_flags
     elif [ -f ./CMakeLists.txt ]; then
         echo "- cmake -D CMAKE_INSTALL_PREFIX=$pkg_prefix $pkg_cmake_flags ."
-        cmake -D CMAKE_INSTALL_PREFIX=$pkg_prefix $pkg_cmake_flags .
+        cmake -D CMAKE_INSTALL_PREFIX=/ $pkg_cmake_flags .
```

Comment:
Note: using `/` rather than `pkg_prefix` because we take advantage of `DESTDIR` to be able to create tarballs that we can store online for speeding up the build process (especially for mobile).